### ISSUE TITLE
[Enhancement][Accessibility]: Horizontal overflow (code blocks)

### ISSUE THREAD

**Issue by JC-DSIT** [alphagov/tech-docs-gem#503]
### Problem
If a content designer adds a code snippet with a long string without spaces the page is displayed with a horizontal scrollbar

## Why is this a problem?

Users will need to scroll in two dimensions to understand the content
Relevant WCAG 2.2 criteria
[1.4.10 Reflow (Level AA)](https://www.w3.org/WAI/WCAG22/Understanding/reflow)

## How could this be improved?
By adding this CSS by default:
code { white-space: pre-wrap; } // OR overflow-x:scroll

**Comment by JC-DSIT** [comment]
After discussing with colleagues in the design system team the said the following:

The design system devs universally disagree with the pre-wrap approach. Code snipped are often contextual to what code is on what line and this means longer lines risk losing context and generally become harder to read. This would technically remove the risk of that criterion failing but make it less usable for everyone else.

It's worth saying as well that the design system website has scrolling code snippets and multiple audits with DAC and this hasn't been raised before. We have an extra feature to allow code snippets to be tab-able, meaning keyboard only users can tab to a code snippet and use the left and right keys to scroll. My guess is that it doesn't fail reflow because it's an interface within a page that does reflow, rather than the whole page not reflowing based on text size and screen size, so a user can still interact with it without issue.~

[This did come up in our last audit](https://github.com/alphagov/govuk-design-system/issues/4007) but after discussing this with DAC and [implementing an additional enhancement](https://github.com/alphagov/govuk-design-system/pull/4034) they agreed that it wasn't a fail.
[#4007 Code blocks: Unnecessary tab stops](https://github.com/alphagov/govuk-design-system/issues/4007)[alphagov/govuk-design-system](https://github.com/alphagov/govuk-design-system) 
[#4034 Add overflow checks to code blocks to dynamically remove `tabindex`](https://github.com/alphagov/govuk-design-system/pull/4034)
